tools/pythonpkg/src/include/duckdb_python/pyconnection.hpp

```diff
@@ -198,6 +214,13 @@ struct DuckDBPyConnection : public std::enable_shared_from_this<DuckDBPyConnecti
 private:
 	PathLike GetPathLike(const py::object &object);
 	unique_lock<std::mutex> AcquireConnectionLock();
+	ScalarFunction CreatePyArrowScalarUDF(const string &name, const py::object &udf, const py::object &parameters,
+	                                      const shared_ptr<DuckDBPyType> &return_type, bool varargs,
+	                                      FunctionNullHandling null_handling,
+	                                      PythonExceptionHandling exception_handling);
+	ScalarFunction CreateScalarUDF(const string &name, const py::object &udf, const py::object &parameters,
```

Comment:
Parameters should really be an Optional<py::list> instead, to ensure correct documentation is generated

Reply:
> Should this also be done for the keyword-only parameters, or are those fine?

We should avoid using py::object in general

Reply:
> Also I thought this would be taken care of by using the proper Optional type annotation in the stubs
> Or are those things not related?

The type stubs are really only for type checking via mypy, pyright, pycharm, etc

The types of the functions themselves show up in our generated documentation on the website, and in the docstrings in ide's, IPython, `help(...)`, etc

Reply:
Alright, I am running into some issues with the Optional wrapper for the return type though.
Since `shared_ptr<DuckDBPyType>` has a type caster, currently we would lose that ability.

The Optional class likely needs a type caster as well, that forwards to the underlying type